Fix/kubernetes fixes and performance

dymbe opened the pull request:
- add caching
- add /ready-endpoint
- fix wrong names in k8s-template

dymbe committed:
fix: add caching

dymbe committed:
fix: wrong names in k8s-template

kstigen commented on geocoder_api/routers/geocoding.py:
```diff
@@ -1,17 +1,22 @@
+@cache(expire=3600)  # Expires after 1 hour
+async def geocode(
+    q: Annotated[str, Query(description="Search query")]
+) -> FeatureCollection:
```
This FeatureCollection does not provide any specification about what type of properties the returned geojson-features have?

Is it possible to also model them, so we can have a full documentation of the returned data?

dymbe committed:
fix: healthcheck correctly

dymbe committed:
Add Photon-models

dymbe committed:
feat: expose additional Photon-query parameters

added lat and lon for regular geocoding, and lang and limit for both regular and reverse geocoding

dymbe commented:
Ended up creating my own models, as adding examples to the `geojson-pydantic`-models was not trivial, and they did not show up correctly in the OpenAPI-spec.

Also ended up exposing and documenting some additional query-parameters

kstigen commented on geocoder_api/routers/geocoding.py:
```diff
@@ -1,38 +1,72 @@
+        }
         res = await client.get(f"{settings.photon_url}/reverse", params=params)
         if res.status_code == 200:
             return res.json()
```
We are not confident that the data we return here actually follows the model that you have specified in FeatureCollection?

dymbe replied:
If we get a 200, it should follow the model defined in FeatureCollection
